### PR TITLE
authentication added

### PR DESCRIPTION
**Pull request by parulgupta2001**
Login and signup pages are added.
Routes and authentication added.
404 page and RequiresAuth page added.

### CODE REVIEW

**Review by therajatg — APPROVED** [review]
looks good

File: src/frontend/components/slider/Slider.js [hunk 1]
```diff
@@ -11,33 +11,32 @@ const properties = {
...
 };
 
-export function Slider() {
+export function Slideshow() {
   const [image, setImage] = useState([]);
 
   useEffect(() => {
     axios.get("/api/videos").then((response) => {
       setImage(response.data.videos);
+      console.log(response.data.videos);
```
Comment (manish591) [inline src/frontend/components/slider/Slider.js]: Remove console if not required.

File: src/frontend/contexts/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext, useReducer } from "react";
+import { authReducer } from "../reducers/auth-reducer";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  // const encodedToken = localStorage.getItem("token");
```
Comment (manish591) [inline src/frontend/contexts/auth-context.js]: Remove commented code

File: src/frontend/pages/login/Login.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      });
+      localStorage.setItem("token", response.data.encodedToken);
+      authDispatch({ type: "ERROR", payload: response.data.encodedToken });
+      navigate("navigate(location.state.from.pathname, { replace: true })");
+    } catch (error) {
+      authDispatch({
+        type: "ERROR",
+        payload: "Wrong credentials, please try again",
+      });
+      console.log(error);
```
Comment (manish591) [inline src/frontend/pages/login/Login.js:27]: can use `console.error` for error messages

**Review by manish591 — APPROVED** [review]
Code is looking good.

File: src/frontend/pages/login/Login.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+  const loginHandler = () => {
+    try {
+      const response = axios.post(`/api/auth/login`, {
+        email: user.email,
+        password: user.password,
+      });
+      localStorage.setItem("token", response.data.encodedToken);
+      authDispatch({ type: "ERROR", payload: response.data.encodedToken });
+      navigate("navigate(location.state.from.pathname, { replace: true })");
```
Comment (manish591) [inline src/frontend/pages/login/Login.js]: can you please check your syntax here? 
It should be 
**navigate('`${location?.state?.from?.pathname}`', { replace: true } )**

**Review by manish591 — COMMENTED** [review]
check

**Pull request merged**